### PR TITLE
Move eslint-config-scratch to GithubActions

### PR DESCRIPTION
**Pull request by delasare**
### Resolves
https://scratchfoundation.atlassian.net/browse/ENG-32

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        - uses: wagoid/commitlint-github-action@v5
+        - name: Use Node.js 18
+          uses: actions/setup-node@v1
+          with:
+            node-version: 18.x
+        - name: Install Dependencies
+          run: npm install
+        - name: Test
+          run: Semantic Release
+        - name: Build
```
Comment (cwillisf) [inline .github/workflows/deploy.yml:21]:
```suggestion
        - name: Test
          run: npm test
        - name: Semantic Release
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Build eslint-config-scratch
+
```
Comment (cwillisf) [inline .github/workflows/deploy.yml:2]: After merging my GHA changes in VM, I discovered that I needed more permissions to let `semantic-release` do its whole job:
https://github.com/scratchfoundation/scratch-vm/blob/cfbc024442e75aea995d1deadb1b6d03cefda7eb/.github/workflows/ci-cd.yml#L12-L16

**Review by cwillisf — CHANGES_REQUESTED** [review]
Haha, glad it was so straight forward!
I think you'll need a couple changes to make the release work, but the rest looks good

**Comment by cwillisf** [comment]
Actually, on second thought, it might be good to have this workflow run on branches & PRs as well. It'll just amount to setup and `npm test` since `semantic-release` will notice it's on a non-deploy branch and do nothing, which seems good for verifying work before merging it.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (delasare) [reply .github/workflows/deploy.yml:21]: ah good catch 🤦

**Comment by delasare** [comment]
I add a 'is master' conditional to the semantic release step

**Review by cwillisf — APPROVED** [review]
👍

**Pull request merged**